spec/services/retrieve_source_file_service_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+require 'hydra/works'
+
+describe Hydra::Derivatives::RetrieveSourceFileService do
+
+  before(:all) do
+    class IndirectContainerObject < ActiveFedora::Base
+      contains "the_source_name"
+    end
+
+    # This uses directly_contains (inherited from Hydra::PCDM::ObjectBehavior)
+    class DirectContainerObject < Hydra::Works::GenericFile::Base
+    end
+    # If you manually built DirectContainerObject, it would look like this:
+    # class DirectContainerObject < ActiveFedora::Base
+    #
+    #   directly_contains :files, has_member_relation: RDFVocabularies::PCDMTerms.hasFile,
+    #     class_name: "Hydra::PCDM::File"
+    #
+    #   def original_file
+    #     file_of_type(::RDF::URI("http://pcdm.org/OriginalFile"))
+    #   end
+    #
+    #   def thumbnail
+    #     file_of_type(::RDF::URI("http://pcdm.org/ThumbnailImage"))
+    #   end
+    # end
```

Comment:
Move lines 14-27 up above line 12, so it's clearer that they're documentation for that class